tests failing with pint 0.21

I noticed the weldx tests failing in the ASDF downstream CI for a recent PR:
https://github.com/asdf-format/asdf/actions/runs/4862963112/jobs/8673553574#step:10:526

I ran some of the tests with ASDF 2.15 and a fresh development install of weldx and saw similar errors (traceback for one test below). They appear to be related to the newly release pint 0.21.

I think this is related to:
https://github.com/hgrecco/pint/pull/1594
not handling list inputs (like is used in the weldx geometry submodule).

```bash
____________________________ test_shape_reflection _____________________________
weldx/tests/test_geometry.py:1735: in test_shape_reflection
    shape_reflection_test_case([2, 1], np.linalg.norm([2, 1]))
weldx/tests/test_geometry.py:1679: in shape_reflection_test_case
    shape = default_test_shape()
weldx/tests/test_geometry.py:1472: in default_test_shape
    arc_segment = geo.ArcSegment.construct_with_points(
../../../.pyenv/versions/3.10.6/envs/weldx_numpy_error/lib/python3.10/site-packages/pint/registry_helpers.py:362: in wrapper
    return func(*args, **kwargs)
weldx/geometry.py:671: in construct_with_points
    return cls(points, arc_winding_ccw)
../../../.pyenv/versions/3.10.6/envs/weldx_numpy_error/lib/python3.10/site-packages/pint/registry_helpers.py:362: in wrapper
    return func(*args, **kwargs)
weldx/geometry.py:572: in __init__
    series = self._update_internals_and_get_series()
weldx/geometry.py:627: in _update_internals_and_get_series
    sign = -1 if np.cross([1, 0], diff) < 0 else 1
<__array_function__ internals>:200: in cross
    ???
../../../.pyenv/versions/3.10.6/envs/weldx_numpy_error/lib/python3.10/site-packages/pint/facets/numpy/quantity.py:64: in __array_function__
    return numpy_wrap("function", func, args, kwargs, types)
../../../.pyenv/versions/3.10.6/envs/weldx_numpy_error/lib/python3.10/site-packages/pint/facets/numpy/numpy_func.py:1004: in numpy_wrap
    return handled[name](*args, **kwargs)
../../../.pyenv/versions/3.10.6/envs/weldx_numpy_error/lib/python3.10/site-packages/pint/facets/numpy/numpy_func.py:772: in implementation
    a = _base_unit_if_needed(a)
../../../.pyenv/versions/3.10.6/envs/weldx_numpy_error/lib/python3.10/site-packages/pint/facets/numpy/numpy_func.py:734: in _base_unit_if_needed
    if a._is_multiplicative:
E   AttributeError: 'list' object has no attribute '_is_multiplicative'
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Release Notes
 
+## 0.6.7 (unreleased)
+
+### Fixes
+
+- fix compatibility with `pint=0.21` \[{pull}`876`\] .
+
 ## 0.6.6 (19.04.2023)
 
 Version `0.6.6` is a compatibility release for `xarray>=2023.4.0` .

--- a/weldx/geometry.py
+++ b/weldx/geometry.py
@@ -624,8 +624,8 @@ class ArcSegment(DynamicShapeSegment):
         self._radius = Q_(np.linalg.norm(diff.m), self._points.u)
 
         expr = "(x*cos(a+s*w)+y*sin(a+s*w))*r + o"
-        sign = -1 if np.cross([1, 0], diff) < 0 else 1
-        a = np.arccos(np.dot([1, 0], diff) / self._radius) * sign
+        sign = -1 if np.cross([1, 0], diff.m) < 0 else 1
+        a = np.arccos(np.dot([1, 0], diff.m) / self._radius.m) * sign
 
         params = dict(
             x=Q_([1, 0, 0], ""),

--- a/weldx/tests/_helpers.py
+++ b/weldx/tests/_helpers.py
@@ -4,6 +4,7 @@ from typing import Any
 
 import numpy as np
 import pint
+from pkg_resources import get_distribution
 
 from weldx.constants import Q_
 from weldx.geometry import _vector_is_close as vector_is_close
@@ -127,4 +128,9 @@ def matrix_is_close(mat_a, mat_b, abs_tol=1e-9) -> bool:
 
     if mat_a.shape != mat_b.shape:
         return False
-    return np.all(np.isclose(mat_a, mat_b, atol=abs_tol)).__bool__()
+
+    atol_unit = 1.0
+    if get_distribution("pint").version >= "0.21":
+        atol_unit = mat_b.u
+
+    return np.all(np.isclose(mat_a, mat_b, atol=abs_tol * atol_unit)).__bool__()

--- a/weldx/tests/_helpers.py
+++ b/weldx/tests/_helpers.py
@@ -130,7 +130,7 @@ def matrix_is_close(mat_a, mat_b, abs_tol=1e-9) -> bool:
         return False
 
     atol_unit = 1.0
-    if get_distribution("pint").version >= "0.21":
+    if isinstance(mat_b, pint.Quantity) and get_distribution("pint").version >= "0.21":
         atol_unit = mat_b.u
 
     return np.all(np.isclose(mat_a, mat_b, atol=abs_tol * atol_unit)).__bool__()

--- a/weldx/tests/transformations/_util.py
+++ b/weldx/tests/transformations/_util.py
@@ -3,6 +3,7 @@ from __future__ import annotations
 from typing import Any, Union
 
 import numpy as np
+from pkg_resources import get_distribution
 from xarray import DataArray
 
 import weldx.transformations as tf
@@ -77,7 +78,13 @@ def check_coordinate_system(
         lcs.orientation, orientation_expected, positive_orientation_expected
     )
 
-    assert np.allclose(lcs.coordinates.data, coordinates_expected, atol=1e-9)
+    atol_unit = 1.0
+    if get_distribution("pint").version >= "0.21":
+        atol_unit = coordinates_expected.u
+
+    assert np.allclose(
+        lcs.coordinates.data, coordinates_expected, atol=1e-9 * atol_unit
+    )
 
 
 def check_cs_close(lcs_0, lcs_1):

